phoenix-core/src/main/java/org/apache/phoenix/coprocessor/MetaDataEndpointImpl.java

```diff
@@ -630,6 +641,13 @@ private PTable getTable(RegionScanner scanner, long clientTimeStamp, long tableT
                 if (statsHTable != null) statsHTable.close();
             }
         }
+        // sort columns by ordinal position
```

Comment:
I had to add this because AlterTableIT.testDropIndexedColumn was failing for local indexes. 
In  PTableImpl.init it looks like  allColumns is sorted, but pkColumns is not sorted. 
Should I just move this into PTableImpl so that PK columns are sorted?